File: Problems/boj_바이러스.py
#https://www.acmicpc.net/problem/2606
#BFS

from collections import deque

n = int(input())
e = int(input())
graph = [[] for _ in range(n+1)]
visited = [False]*(n+1)
for i in range(e):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

q = deque([1])
visited[1] = True

while q:
    now = q.popleft()
    for i in graph[now]:
        if not visited[i]:
            visited[i] = True
            q.append(i)

print(visited.count(True)-1)

File: Problems/cpp/boj_바이러스.cpp
//https://www.acmicpc.net/problem/2606
//DFS, BFS

#include <iostream>
#include <vector>

using namespace std;

int n, e, cnt=0;
vector<vector<int>> graph;
vector<bool> visit;

void dfs(int start){
    cnt++;
    visit[start] = true;

    for (const int& v : graph[start]){
        if (!visit[v]){
            dfs(v);
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    cin >> n >> e;

    graph.resize(n+1);
    visit.resize(n+1, false);
    int a, b;
    for (int i=0; i<e; i++){
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    dfs(1);

    cout << cnt-1;
}